mllib/src/main/scala/org/apache/spark/ml/param/params.scala

```diff
@@ -435,6 +435,43 @@ class BooleanParam(parent: String, name: String, doc: String) // No need for isV
   }
 }
 
+/**
+ * :: DeveloperApi ::
+ * Specialized version of `Param[String]` for Java.
+ */
+@DeveloperApi
+class StringParam(parent: Params, name: String, doc: String, isValid: String => Boolean)
+  extends Param[String](parent, name, doc, isValid) {
+
+  private var options: Option[Array[String]] = None
+
+  def this(parent: Params, name: String, doc: String) =
+    this(parent, name, doc, ParamValidators.alwaysTrue)
+
+  /** construct a StringParam with limited options (case-insensitive) */
+  def this(parent: Params, name: String, doc: String, options: Array[String]) = {
+    this(parent, name, doc + s" Supported options (case-insensitive): ${options.mkString(", ")}.",
```

Comment:
Is this tested somewhere?

Reply:
It's currently not. Do you mean we should test the `doc` correctly include all the options? or there's any special case that should be covered.

Reply:
I missed one additional line when highlighting, sorry for that :)
I meant to test the case-insensitive validation in the next line.

Doc could be tested too, I think it's rather a nice to have than a must.

Reply:
I see. We have embedded the test into `checkParams` but that only checks the legal cases. 
I'll add a unit test for illegal cases, and we can check the doc and exception message.